### ISSUE TITLE
개인별 모델 baseline 구축 및 실험 결과 공유_윤정

### ISSUE THREAD

**Issue by nimnusmik** [AIBootcamp14/upstageailab-ml-competition-ml-5#73]
### 실험 1  
- `자치구`, `브랜드등급`을 `One-Hot Encoding` 처리
- 실제로 모델에 적용해보니 의미 없을것 같은 자치구 feature의 중요도가 높게 나와버림
- 하이퍼파라미터는 ChatGPT 코드 참고하여 `선형회귀`, `랜덤포레스트`, `XGBoost` 사용
```
linear = LinearRegression(fit_intercept=True, n_jobs=-1, positive=False)
rf = RandomForestRegressor(n_estimators=300, max_depth=15, bootstrap=True, random_state=seed, n_jobs=-1)
xgb = XGBRegressor(n_estimators=300, max_depth=30, learning_rate=0.1, subsample=0.8, colsample_bytree=0.8, random_state=seed, n_jobs=-1)
```
- 결과
    <img width="304" height="62" alt="Image" src="https://github.com/user-attachments/assets/2f726870-c1a3-4b12-bcff-0b4e67ba7e90" />
    <img width="547" height="425" alt="Image" src="https://github.com/user-attachments/assets/2c513c02-9191-4753-bd40-b0635b973935" />
    <img width="545" height="427" alt="Image" src="https://github.com/user-attachments/assets/d49a777f-b57b-443b-b170-baac7183c98a" />

### 실험 2
- `자치구`는 `강남3구여부`가 있으니 사용하지 않고 `자치구_법정동` 파생변수를 가지고 `Encoding` 처리
- `Label Encoding`은 모델이 숫자의 크기(순서)에 의미가 있다고 오해할 수 있다고 해서 최근 10년간 계약건 중 `자치구_법정동`으로 그룹화하여 평균 평당가 기준으로 정렬하여 label 추가
- 이 경우 `target`값을 사용하여 labeling 했기 때문에 변수 중요도가 높아져버림
- 결과 (하이퍼파라미터는 실험1과 동일)
    <img width="310" height="65" alt="Image" src="https://github.com/user-attachments/assets/cb02c326-76da-4f4b-aa7d-a176c0d988ca" />
    <img width="673" height="500" alt="Image" src="https://github.com/user-attachments/assets/7af9efe5-36c9-4e57-9920-d33f859ef1d8" />
    <img width="667" height="499" alt="Image" src="https://github.com/user-attachments/assets/67cefc23-a751-4976-9463-490205fdede7" />

**Comment by yjjang06** [comment]
### 실험3
- k-fold 5로 설정하여 lightGBM 모델링 해보았으나 XGBoost성능이 더 좋게 나옴
    ```
    lgb_model = lgb.LGBMRegressor(
        n_estimators=3000,
        learning_rate=0.01,
        max_depth=15,
        num_leaves=128,
        subsample=0.6,
        colsample_bytree=0.8,
        reg_alpha=1,
        reg_lambda=1,
        random_state=seed,
        n_jobs=-1
    )
    xgb = XGBRegressor(
        n_estimators=3000,
        learning_rate=0.01,
        max_depth=15,
        subsample=0.6,
        colsample_bytree=0.8,
        reg_alpha=1,
        reg_lambda=1,
        random_state=seed,
        n_jobs=-1
    )
    ```
    |모델|RMSE|
    |----|----|
    |lightGBM|6616|
    |XGBoost|5483|
- 두 모델 다 절편이 크게 나와서 절편을 작게 하는 방법을 찾아보고 다시 실험할 예정

#### lightGBM
<img width="579" height="414" alt="Image" src="https://github.com/user-attachments/assets/853a06bd-15ae-4c97-b841-e0966145ebc0" />

#### XGBoost
<img width="553" height="407" alt="Image" src="https://github.com/user-attachments/assets/0f454ee0-88f5-4028-86da-21897c57c33a" />
